File: 卡码/代码随想录/Go/贪心/01-简单/LeetCode_0406.go
// 406. 根据身高重建队列
package main

import (
	"sort"
	"slices"
)

func reconstructQueue(people [][]int) [][]int {
	q:=make([][]int,len(people))
	sort.Slice(people,func(i,j int) bool{
		if people[i][0]==people[j][0] {
			return people[i][1]<people[j][1]
		}else{
			return people[i][0]>people[j][0]
		}
	})
	for _,p:=range people {
		q=slices.Insert(q,p[1],p)
	}
	return q[:len(people)]
}

File: 卡码/代码随想录/JS/贪心/02-中等/LeetCode_0406.js
//406. 根据身高重建队列
const reconstructQueue = function(people) {
	const q=new Array();
	people.sort((p1,p2)=>{
		if(p1[0]==p2[0]) return p1[1]-p2[1];
		else return p2[0]-p1[0];
	});
	for(let p of people){
		q.splice(p[1],0,p);
	}
	return q;
};